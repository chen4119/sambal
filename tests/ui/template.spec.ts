import {
    template
} from "../../src/ui/template";


describe("template", () => {

    it('render array', async () => {
        const listItems = [
            "<li>one</li>",
            "<li>two</li>",
            "<li>three</li>"
        ]
        const html = await template`
            <ul>
                ${listItems}
            </ul>
        `;
        expect(html).toMatchSnapshot();
    });

    it('render nothing for null', async () => {
        const listItems = null;
        const html = await template`
            <ul>
                ${listItems}
            </ul>
        `;
        expect(html).toMatchSnapshot();
    });

    it('render async', async () => {
        const renderItem = async () => template`<li>two</li>`;
        const html = await template`
            <ul>
                ${renderItem()}
            </ul>
        `;
        expect(html).toMatchSnapshot();
    });

    it('nested template', async () => {
        const itemFragment = template`
            <li>
                <b>Hello world</b>
            </li>
        `;

        const html = await template`
            <ul>
                ${itemFragment}
            </ul>
        `;

        console.log(html);
    });

});