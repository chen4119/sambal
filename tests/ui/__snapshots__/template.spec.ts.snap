// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template nested template 1`] = `
"
            <ul>
                
            <li>
                <b>Hello world</b>
            </li>
        
            </ul>
        "
`;

exports[`template render array 1`] = `
"
            <ul>
                <li>one</li><li>two</li><li>three</li>
            </ul>
        "
`;

exports[`template render async 1`] = `
"
            <ul>
                <li>two</li>
            </ul>
        "
`;

exports[`template render nothing for null 1`] = `
"
            <ul>
                
            </ul>
        "
`;
